src/DynamoCoreWpf/ViewModels/PackageManager/PackageViewModel.cs

```diff
@@ -233,24 +256,125 @@ private void Uninstall()
             }
             return false;
         }
 
-        private void GoToRootDirectory()
+        // Loads a built-in package that was previously set as Unloaded
+        private void Load()
+        {
+            var dynModel = dynamoViewModel.Model;
+            var packageLoader = dynModel.GetPackageManagerExtension().PackageLoader;
+            var conflicts = packageLoader.LocalPackages.ToList().Where(x => x.Name == Model.Name && x != Model);
+            bool hasConflictsWithLoadedAssemblies = conflicts.Any(x => x.InUse(dynamoViewModel.Model));
+
+            if (conflicts.Any())
+            {
+                string conflictsMsg = string.Join(",", conflicts.Select(x => x.Name + " " + x.VersionName));
+
+                string message = hasConflictsWithLoadedAssemblies ? 
+                        string.Format(Resources.MessageLoadBuiltInPackageWithRestart,
+                        conflictsMsg, Model.Name + " " + Model.VersionName) 
+                        :
+                        string.Format(Resources.MessageLoadBuiltInPackage,
+                        conflictsMsg, Model.Name + " " + Model.VersionName);
+
+                var dialogResult = MessageBoxService.Show(message,
+                    Resources.CannotDownloadPackageMessageBoxTitle,
+                        MessageBoxButton.OKCancel,
+                        MessageBoxImage.Exclamation);
+
+                if (dialogResult == MessageBoxResult.Cancel) return;
+            }
+
+            try
+            {
+                foreach (var package in conflicts)
+                {
+                    package.UninstallCore(dynModel.CustomNodeManager, packageLoader, dynModel.PreferenceSettings);
+                }
+
+                if (!hasConflictsWithLoadedAssemblies)
+                {
+                    try
+                    {
+                        packageLoader.TryLoadPackageIntoLibrary(Model);
+                        Model.LoadState.SetAsLoaded();
+                    } 
+                    catch 
+                    {
+                        Model.MarkForLoad(dynModel.PreferenceSettings);
+                    }
+                }
+                else
+                {
+                    Model.MarkForLoad(dynModel.PreferenceSettings);
+                }
+            }
+            catch (Exception e) 
+            {
+                Console.WriteLine(e.Message);
+            }
+            finally
+            {
+                NotifyLoadStatePropertyChanged();
+                foreach (var package in dynamoViewModel.PreferencesViewModel.LocalPackages.Where(x => conflicts.Contains(x.Model)))
+                {
+                    package.NotifyLoadStatePropertyChanged();
+                }
+            }
+        }
+
+        // Cancels the ScheduledForLoad status on a built-in package
+        private void UnmarkForLoad()
+        {
+            var dynModel = dynamoViewModel.Model;
```

Comment:
this seems like it could be factored out into a private helper

Reply:
good idea